pkg/library/libraryinputresources/resource.go

```diff
@@ -63,7 +70,24 @@ func LenientResourcesFromDirRecursive(location string) ([]*Resource, error) {
 	return currResourceList, errors.Join(errs...)
 }
 
-func ResourceFromFile(location, fileTrimPrefix string) (*Resource, error) {
+func findGVR(resources map[schema.GroupVersion]*metav1.APIResourceList, gvk schema.GroupVersionKind) (*schema.GroupVersionResource, error) {
+	apiResourceList, ok := resources[gvk.GroupVersion()]
+	if !ok {
+		return nil, fmt.Errorf("failed to find api resource list for gvk %s", gvk)
+	}
+	for _, apiResource := range apiResourceList.APIResources {
+		if apiResource.Kind == gvk.Kind {
+			return &schema.GroupVersionResource{
+				Group:    gvk.Group,
+				Version:  gvk.Version,
+				Resource: apiResource.Name,
+			}, nil
+		}
```

Comment:
There can be more than one result.  Need to make a list of all results and return an error if there is more than one hit.